bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -400,7 +400,12 @@ async fn extension_loop_active(
                     match event {
                         Event::Metric(event) => {
                             debug!("Metric event: {:?}", event);
-                        }
+                        },
+                        Event::OutOfMemory => {
+                            let mut p = invocation_processor.lock().await;
+                            p.on_out_of_memory_error();
+                            drop(p);
```

Comment:
do we need this explicit drop?
They are all over the place and I think they are useless since it is done automatically at the end of the block

Reply:
Maybe not on the ones where we have it at the end, we can improve this later